### ISSUE TITLE
about the loss functional

### ISSUE THREAD

**Issue by wangshihuang** [justchenhao/STANet#42]
您好，我想问问关于loss函数，我在CDFA_model看到
            self.L_s[self.L_s == 1] = -1  # change
            self.L_s[self.L_s == 0] = 1  # no change
在loss.py里看到
class BCL(nn.Module):
    """
    batch-balanced contrastive loss
    no-change，1
    change，-1
    """

    def __init__(self, margin=2.0):
        super(BCL, self).__init__()
        self.margin = margin

    def forward(self, distance, label):
        label[label == 255] = 1
        mask = (label != 255).float()
        distance = distance * mask
        pos_num = torch.sum((label == 1).float()) + 0.0001
        neg_num = torch.sum((label == -1).float()) + 0.0001

        loss_1 = torch.sum((1 + label) / 2 * torch.pow(distance, 2)) / pos_num
        loss_2 = torch.sum((1 - label) / 2 *
                           torch.pow(torch.clamp(self.margin - distance, min=0.0), 2)
                           ) / neg_num
        loss = loss_1 + loss_2
        return loss
为什么要这样写啊，或者说不改变label的值，就是原来的0和1，loss函数要怎么修改，margin的值怎么写；
还有loss函数forward下面第一句label[label == 255] = 1是不是没有用啊；
        mask = (label != 255).float()
        distance = distance * mask
这个好像也没用
@justchenhao

**Comment by wangshihuang** [comment]
pos_num在原始label上，应该是0的个数还是1的个数
